lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -258,6 +258,9 @@ module InternalConstants
     config_param :zone, :string, :default => nil
     config_param :vm_id, :string, :default => nil
     config_param :vm_name, :string, :default => nil
+    # Kubernetes-specific parameters.
+    config_param :k8s_cluster_name, :string, :default => nil
```

Comment:
My only user experience concern:

With the current implementation, it requires users to set `enable_metadata_agent` to true in Logging Agent, but not install metadata agent (or somehow makes sure metadata agent does not return anything given that `local_resource_id`). So that the customized `k8s_cluster_name` and `k8s_location` would take effect. Would that be a bit confusing?

Alternatively, we could detect that `k8s_cluster_name` and `k8s_location` are set and execute specific logic regardless of the value of `enable_metadata_agent`. Something like:

```
      if local_resource_id
        if @enable_metadata_agent
          # Exactly what we were doing before here.
        elsif @k8s_cluster_name && @k8s_location
          constructed_k8s_resource = construct_k8s_resource_locally(
            local_resource_id)
          resource = constructed_k8s_resource if constructed_k8s_resource
        end
      end
```

Thoughts?

Reply:
Per clarification offline, the expected way of doing this is always installing Metadata Agent and enabling the communication from Logging Agent to Metadata Agent. Metadata Agent has a similar mechanism to specify @k8s_cluster_name and @k8s_location. This change is just for fallbacks. So the current implementation makes sense.

Reply:
Can we enrich the comment of this parameter a bit (same with `k8s_location`)? Something like:

`# Customized Kubernetes cluster name. Stackdriver Metadata Agent and Logging Agent detect Kubernetes cluster names in GKE automatically. For other platforms this has to be specified explicitly.`

?

Reply:
I've added something similar without mentioning GKE and other platforms. PTAL.